app/controllers/invite_requests_controller.rb

```diff
@@ -10,17 +10,20 @@ def index
 
   # GET /invite_requests/1
   def show
-    @invite_request = InviteRequest.find_by(email: params[:email])
-
-    if @invite_request.present?
-      @position_in_queue = @invite_request.position
+    if params[:email].blank?
+      @email_blank = true
```

Comment:
Definitely not sure it's better. But, in theory, it should be possible to use an early return here, something like (quick example, please don't trust me on the exact syntax):
```rb
if params[:email].blank?
  return respond_to do |format|
    format.html { render "invite_requests/_no_email" }
    format.js { render "invite_requests/_no_email" }
  end
end
```
Pointing directly at the right "final" erb/js templates, without the need to pass around that email_blank variable or modify existing templates.

Reply:
I think this might not work for the javascript version :thinking: